dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -569,8 +570,10 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
         }
         oldWorkflow.setForumUrl(newWorkflow.getForumUrl());
         oldWorkflow.setTopicManual(newWorkflow.getTopicManual());
+        oldWorkflow.setTopicAI(newWorkflow.getTopicAI());
 
-        if (!Objects.equals(oldWorkflow.getMode(), WorkflowMode.HOSTED)) {
+        if (!Objects.equals(oldWorkflow.getMode(), WorkflowMode.HOSTED)
+                || (Objects.equals(oldWorkflow.getMode(), WorkflowMode.HOSTED) && newWorkflow.getTopicSelection() != TopicSelection.AUTOMATIC)) {
```

Comment:
This `if` copies the selection mechanism if the destination workflow isn't hosted, or is hosted and the source workflow selection mechanism isn't automatic.  Why do we set the destination workflow selection mechanism in some cases, but not others?

Reply:
I believe the idea behind this is that a hosted workflow cannot have an automatic topic selection because a hosted workflow has no source control. I'll add a comment to clarify this